vendors/espressif/boards/esp32/ports/ble/nimble/iot_ble_hal_gap.c

```diff
@@ -535,12 +535,13 @@ BTStatus_t prvBTSetAdvData( uint8_t ucAdapterIf,
                             BTUuid_t * pxServiceUuid,
                             size_t xNbServices )
 {
+    const size_t c_max_uuid_per_adv_message = 2;
     struct ble_hs_adv_fields fields;
     const char * name;
     int xESPStatus;
-    ble_uuid16_t uuid16 = { 0 };
-    ble_uuid32_t uuid32 = { 0 };
-    ble_uuid128_t uuid128 = { 0 };
+    ble_uuid16_t uuid16[c_max_uuid_per_adv_message];
```

Comment:
You don't need the array and `c_max_uuid_per_adv_message` as there can be at max one uuid of each type.